### PR TITLE
new Makefile

### PR DESCRIPTION
**Pull request by maaike-slt**
update `Makefile` with standard variable and better progress bar

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,6 +18,12 @@ CFLAGS =	-Wall -Werror -Wextra
 CFLAGS +=	-g
 # CFLAGS +=	-fsanitize=address
 
+ASMFLAGS =	-MMD -MP
+ASMINC =	-I./inc/
+
+LDFLAGS =	-MMD -MP
```
Comment (maaike-slt) [inline Makefile]: not supposed to have double `-MMD -MP`

File: Makefile [hunk 2]
```diff
@@ -67,50 +79,59 @@ OBJ = $(addprefix $(OBJ_PATH), $(OBJ_NAME))
...
+	fi
 
 launch:
 	$(call progress_bar)
 
-$(NAME):	$(OBJ) $(LIBNAME)
-	$(CC) $(CFLAGS) $(OBJ) $(LIBNAME) -o $(NAME) $(LIBREADLINE)
-	@printf "\n$(B)$(MAG)$(NAME) compiled$(D)\n"
+$(NAME):	$(LOCAL_LIB) $(OBJ)
+	$(CC) $(CFLAGS) $(OBJ) $(LOCAL_LIB) $(LDLIBS) -o $(NAME)
```
Comment (maaike-slt) [inline Makefile]:
```suggestion
	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJ) $(LOCAL_LIB) $(LDLIBS) -o $(NAME)
```

File: libft/Makefile [hunk 3]
```diff
@@ -20,6 +20,8 @@ CFLAGS =	-Wall -Werror -Wextra
 # CFLAGS +=	-fsanitize=address
 # CFLAGS +=	-Wsuggest-attribute=const
 
+LKFLAGS =	-MMD -MP
```
Comment (maaike-slt) [inline libft/Makefile]: need `LDFLAGS` and other standard `FLAGS`

File: libft/Makefile [hunk 4]
```diff
@@ -87,30 +97,38 @@ OBJ =				$(addprefix $(OBJ_PATH), $(OBJ_NAME))
...
 
 $(NAME):	$(OBJ)
 	$(AR) $(NAME) $(OBJ)
+	@$(eval name = 1)
 
 $(OBJ_PATH)%.o: $(SRC_PATH)%.c
 	@mkdir -p $(dir $@)
-	$(CC) $(CFLAGS) -I$(INC) -c $< -o $@
-	@printf "$(B)$(GRE)█$(D)"
+	$(CC) $(CFLAGS) $(LKFLAGS) -I$(INC) -c $< -o $@
```
Comment (maaike-slt) [inline libft/Makefile]:
```suggestion
	$(CC) $(CFLAGS) $(LDFLAGS) -I$(INC) -c $< -o $@
```

File: Makefile (hunk 1 above)
Comment (Alexdelia) [reply Makefile]: fixed in f7323dd

**Pull request merged**